contracts/modules/LoanSettings/LoanSettings.sol

```diff
@@ -93,6 +95,99 @@ contract LoanSettings is State, InterestHandler, LoanSettingsEvents {
         }
     }
 
+    function migrateLoanParamsList(
+        address owner,
+        uint256 start,
+        uint256 count)
+        external
+        onlyGuardian
+    {
+        EnumerableBytes32Set.Bytes32Set storage set = userLoanParamSets[owner];
+        uint256 end = start.add(count).min256(set.length());
+        if (start >= end) {
+            return;
+        }
+        count = end-start;
+
+        bytes32 loanParamId;
+        LoanParams memory loanParamsLocal;
+
+        for (uint256 i = start; i < end; ++i) {
+            loanParamsLocal = loanParams[set.get(i)];
+            loanParamId = getLoanParam(
+                loanParamsLocal.loanToken,
+                loanParamsLocal.collateralToken,
+                loanParamsLocal.maxLoanTerm == 0 // isTorqueLoan
+                    ? true
+                    : false
+            );
+            loanParamsIds[loanParamId] = loanParamsLocal.id;
+        }
+    }
+
+    function getLoanParam(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan)
+        pure
+        internal
+        returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(
+                    loanToken,
+                    collateralToken,
+                    isTorqueLoan
+                ));
+    }
+
+    // This function intends to be PUBLIC so that anyone can create params if loanToken and collateralToken are approved by the protocol
+    function createDefaultParams(
+        address loanToken,
+        address collateralToken,
+        bool isTorqueLoan)
+        external
+    {
+        // requires loanToken approved
+        require(supportedTokens[loanToken], "invalid");
+        // requires collateralToken approved
+        require(supportedTokens[collateralToken], "invalid");
+        // requires there is a pricefeed
+        require(IPriceFeeds(priceFeeds).pricesFeeds(collateralToken) != address(0), "invalid");
```

Comment:
having unique error messages helps a lot with debugging potential issues in the future and communicates why an error occurred better.